modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -1668,6 +1733,45 @@ public String updateDescription(String descriptionDetails, String user) throws P
     }
 
     /**
+     * Delete the process related artifacts : Documents, BPMN model, Flow chart , Process text
+     *
+     * @param processName
+     * @param processVersion
+     */
+    public void deleteProcessRelatedArtifacts(String processName, String processVersion, String user) throws
+            ProcessCenterException {
+
+        String processResourcePath =
+                ProcessCenterConstants.GREG_PATH + ProcessCenterConstants.PROCESS_ASSET_ROOT + processName + "/"
+                        + processVersion;
+        try {
+            RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+            if (registryService != null) {
+                //delete Process Documents
+                String uploadedDocs = getUploadedDocumentDetails(processResourcePath);
+                JSONArray uploadedDocJSNArray = new JSONArray(uploadedDocs);
+
+                for (int i = 0; i < uploadedDocJSNArray.length(); i++) {
+                    JSONObject deleteDocument = new JSONObject();
+                    deleteDocument.put("processName", processName);
+                    deleteDocument.put("processVersion", processVersion);
+                    deleteDocument.put("removeDocument", uploadedDocJSNArray.get(i));
+                    deleteDocument(deleteDocument.toString(), user);
+                }
+                deleteDocumentResourceCollection(processName, processVersion);
+
+                //delete other associations
+                removeBPMNDiagram(processName, processVersion, user);
+                deleteFlowchart(processName, processVersion, user);
+                deleteProcessText(processName, processVersion);
+            }
+        } catch (Exception e) {
+            String errMsg =
```

Comment:
Thanks @mohanvive for pointing out the issue. Fixed with 98b499b6273a21f443e4edbd49b29672d8740e3b